data_steward/cdr_cleaner/cleaning_rules/aggregate_zip_codes.py

```diff
@@ -0,0 +1,235 @@
+        AND mzc.person_id = obs.person_id
+""")
+
+MODIFY_STATES_QUERY = JINJA_ENV.from_string("""
+WITH unique_state_transforms AS (
+    SELECT DISTINCT
+        person_id, state_observation_id, state,
+        transformed_state
+    FROM `{{project_id}}.{{sandbox_id}}.{{modified_zip_codes_and_states_table}}` mzc
+)
+SELECT
+    obs.* REPLACE (
+        COALESCE(state_vocab.concept_id, obs.value_source_concept_id) AS value_source_concept_id,
+        COALESCE(state_vocab.concept_id, obs.value_as_concept_id) AS value_as_concept_id,
+        COALESCE(state_vocab.concept_code, obs.value_source_value) AS value_source_value
+    )
+FROM `{{project_id}}.{{dataset_id}}.{{obs_table}}` obs
+LEFT JOIN unique_state_transforms mzc
+    ON mzc.state_observation_id = obs.observation_id
+        AND mzc.person_id = obs.person_id
+LEFT JOIN `{{project_id}}.{{pipeline_tables_dataset}}.{{pii_state_vocab}}` state_vocab
+    ON state_vocab.postal_code = mzc.transformed_state
+""")
+
+
+class AggregateZipCodes(BaseCleaningRule):
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'Aggregates generalized zip codes based on first digits.'
+        super().__init__(
+            issue_numbers=['DC1379'],
+            description=desc,
+            affected_datasets=[cdr_consts.CONTROLLED_TIER_DEID],
+            affected_tables=[OBSERVATION],
+            project_id=project_id,
+            dataset_id=dataset_id,
+            sandbox_dataset_id=sandbox_dataset_id,
+            depends_on=[GeneralizeZipCodes, StringFieldsSuppression])
+        # Identifiable information may exist if StringFieldsSuppression rule fails
+
+    def get_query_specs(self, *args, **keyword_args):
+        """
+        Interface to return a list of query dictionaries.
+
+        :returns:  a list of query dictionaries.  Each dictionary specifies
+            the query to execute and how to execute.  The dictionaries are
+            stored in list order and returned in list order to maintain
+            an ordering.
+        """
+        find_zip_codes_and_states_query = dict()
+        find_zip_codes_and_states_query[
+            cdr_consts.QUERY] = FIND_ZIP_CODES_AND_STATES_QUERY.render(
+                project_id=self.project_id,
+                dataset_id=self.dataset_id,
+                obs_table=OBSERVATION,
+                pipeline_tables_dataset=PIPELINE_TABLES,
+                pii_state_vocab=PII_STATE_VOCAB,
+                zip_code_aggregation_map=ZIP_CODE_AGGREGATION_MAP)
+        find_zip_codes_and_states_query[
+            cdr_consts.DESTINATION_TABLE] = self.sandbox_table_for(
+                ZIP_CODES_AND_STATES_TO_MODIFY)
```

Comment:
if changed to use `SANDBOX_QUERY`
`cdr_consts.DESTINATION_TABLE = self.sandbox_table_for(OBSERVATION)`